tests/src/org.hibernate.orm/hibernate-core/6.6.0.Final/src/test/java/org_hibernate_orm/hibernate_core/AbstractHibernateTest.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_hibernate_orm.hibernate_core;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.Persistence;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org_hibernate_orm.hibernate_core.entity.Course;
+import org_hibernate_orm.hibernate_core.entity.CourseMaterial;
+import org_hibernate_orm.hibernate_core.entity.Gender;
+import org_hibernate_orm.hibernate_core.entity.Student;
+import org_hibernate_orm.hibernate_core.entity.Teacher;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+abstract class AbstractHibernateTest {
+
+    private EntityManagerFactory entityManagerFactory;
+    private EntityManager entityManager;
+
+    private Map<String, Long> studentIds = new HashMap<>();
+
+    private Map<String, Long> teacherIds = new HashMap<>();
+
+    private Map<String, Long> courseIds = new HashMap<>();
+
+    @BeforeAll
+    public void init() {
+        Map<String, String> properties = new HashMap<>();
+        properties.put("jakarta.persistence.jdbc.url", getJdbcUrl());
+        properties.put("hibernate.dialect", getHibernateDialect());
+        entityManagerFactory = Persistence.createEntityManagerFactory("StudentPU", properties);
+        entityManager = entityManagerFactory.createEntityManager();
+        prepareTestData();
+    }
+
+    @AfterAll
+    public void close() {
+        entityManager.close();
```

Comment:
Minor: Don't we need null check here (same as you have done [here](https://github.com/oracle/graalvm-reachability-metadata/pull/523/files#diff-7cd6280680c74063a7f29799f5b1a2c0069e8d6180322d53fd3fce03872b8c60R31))

Reply:
I have just copied Hibernate 6.5 tests.